### PR TITLE
Fixing test playbook validations

### PR DESCRIPTION
**Pull request by hod-alpert**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Description:
Test playbook validation:
* Scripts: No validation
* Playbooks: Only checking if tests are configured in the yml file
* Integrations: Checking if the integration has a test configured in conf.json, if not: validating 'No tests' is the the yml file
* TestPlaybooks: Validating that there is a test configuration with that test playbook id.

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: demisto_sdk/commands/common/hook_validations/base_validator.py [hunk 1]
```diff
@@ -130,17 +129,6 @@ def _load_conf_file(self):
...
-        Checks if a file (playbook or integration) has a TestPlaybook and if the TestPlaybook is configured in conf.json
-        And prints an error message accordingly
-        """
-        file_type = self.structure_validator.scheme_name
-        tests = self.current_file.get('tests', [])
-        if not self.yml_has_test_key(tests, file_type):
-            return False
-        return self.tests_registered_in_conf_json_file(tests)
-
     def tests_registered_in_conf_json_file(self, test_playbooks: list) -> bool:
```
Comment (yaakovi) [inline demisto_sdk/commands/common/hook_validations/base_validator.py:132]: improve function name:
```suggestion
    def are_tests_registered_in_conf_json_file(self, test_playbooks: list) -> bool:
```

File: demisto_sdk/commands/common/hook_validations/base_validator.py [hunk 2]
```diff
@@ -155,26 +143,39 @@ def tests_registered_in_conf_json_file(self, test_playbooks: list) -> bool:
...
-                f'Please add\n{missing_test_configurations}\nto {self.CONF_PATH} path under \'tests\' key.'
-            print_error(error_message)
-            return False
+        # Test playbook case
+        if 'TestPlaybooks' in self.file_path and file_type == 'playbook':
+            is_configured_test = any(test_config for test_config in conf_json_tests if
+                                     find_test_match(test_config, file_type, test_playbook_id=content_item_id))
+            missing_test_configurations = json.dumps([
+                {'playbookID': content_item_id}
+            ], indent=4).strip('[]')
```
Comment (yaakovi) [inline demisto_sdk/commands/common/hook_validations/base_validator.py:157]: why are you passing a list if you remove it afterward? to get the tab? seems redundant

File: demisto_sdk/commands/common/hook_validations/base_validator.py [hunk 3]
```diff
@@ -155,26 +143,39 @@ def tests_registered_in_conf_json_file(self, test_playbooks: list) -> bool:
...
+            if not is_configured_test:
+                error_message = \
+                    f'The TestPlaybook {content_item_id} is not registered in {self.CONF_PATH} file.\n' \
+                    f'Please add\n{missing_test_configurations}\nto {self.CONF_PATH} path under \'tests\' key.'
```
Comment (yaakovi) [inline demisto_sdk/commands/common/hook_validations/base_validator.py:161]: suggest adding relevant integrations as well

File: demisto_sdk/commands/common/hook_validations/base_validator.py [hunk 4]
```diff
@@ -155,26 +143,39 @@ def tests_registered_in_conf_json_file(self, test_playbooks: list) -> bool:
...
-            error_message = \
-                f'The following TestPlaybooks are not registered in {self.CONF_PATH} file.\n' \
-                f'Please add\n{missing_test_configurations}\nto {self.CONF_PATH} path under \'tests\' key.'
-            print_error(error_message)
-            return False
+        # Test playbook case
+        if 'TestPlaybooks' in self.file_path and file_type == 'playbook':
+            is_configured_test = any(test_config for test_config in conf_json_tests if
+                                     find_test_match(test_config, file_type, test_playbook_id=content_item_id))
+            missing_test_configurations = json.dumps([
```
Comment (yaakovi) [inline demisto_sdk/commands/common/hook_validations/base_validator.py:155]: move inside the condition

File: demisto_sdk/commands/common/hook_validations/base_validator.py [hunk 5]
```diff
@@ -155,26 +143,39 @@ def tests_registered_in_conf_json_file(self, test_playbooks: list) -> bool:
...
+                    f'Please add\n{missing_test_configurations}\nto {self.CONF_PATH} path under \'tests\' key.'
+                print_error(error_message)
+                return False
+        # Integration case
+        elif file_type == 'integration':
+            is_configured_test = any(
+                test_config for test_config in conf_json_tests if find_test_match(test_config,
+                                                                                  file_type,
+                                                                                  integration_id=content_item_id))
+            missing_test_configurations = json.dumps([
```
Comment (yaakovi) [inline demisto_sdk/commands/common/hook_validations/base_validator.py:170]: move inside the condition

File: demisto_sdk/commands/common/hook_validations/playbook.py [hunk 6]
```diff
@@ -43,6 +43,18 @@ def is_valid_playbook(self, is_new_playbook: bool = True, validate_rn: bool = Tr
...
+        """
+        Checks if the playbook has a TestPlaybook and if the TestPlaybook is configured in conf.json
+        And prints an error message accordingly
+        """
+        if 'TestPlaybooks' in self.file_path:
+            return True
+
+        file_type = self.structure_validator.scheme_name
+        tests = self.current_file.get('tests', [])
+        return self.yml_has_test_key(tests, file_type)
```
[the remark this answers is not in the document]
Comment (hod-alpert) [reply demisto_sdk/commands/common/hook_validations/playbook.py:56]: That's not the build's process

[the remark this answers is not in the document]
Comment (yaakovi) [reply demisto_sdk/commands/common/hook_validations/playbook.py:56]: enhancement for another day

**Pull request merged**